### PR TITLE
Add clear-filter-X button to datasets page search box

### PR DESCRIPTION
**Pull request by adamabeshouse**
image

### CODE REVIEW

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx [hunk 1]
```diff
@@ -249,7 +250,7 @@ function getDownloadObject<T>(columns: Column<T>[], rowData: T) {
 }
 
 export class LazyMobXTableStore<T> {
-    public filterString: string | undefined;
+    @observable public filterString: string | undefined;
```
Comment (adamabeshouse) [inline src/shared/components/lazyMobXTable/LazyMobXTable.tsx:253]: I wonder why this wasnt observable before... hopefuly nothing breaks

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx [hunk 2]
```diff
@@ -954,10 +963,32 @@ export default class LazyMobXTable<T> extends React.Component<
                                 className="form-control tableSearchInput"
                                 style={{ width: 200 }}
                             />
+                            {this.props.showFilterClearButton &&
```
Comment (alisman) [inline src/shared/components/lazyMobXTable/LazyMobXTable.tsx:966]: maybe we we swap it with the magnifying glass instead of having two next to each other?

Comment (adamabeshouse) [reply src/shared/components/lazyMobXTable/LazyMobXTable.tsx:966]: good idea, updated

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx (hunk 1 above)
Comment (alisman) [reply src/shared/components/lazyMobXTable/LazyMobXTable.tsx:253]: i actually feel like there was a reason.  let me test it out and think on it.

Comment (alisman) [reply src/shared/components/lazyMobXTable/LazyMobXTable.tsx:253]: @adamabeshouse unless there is a strong reason to make it observable (like, we are fixing something known to be broken), lets just leave this as it was.

Comment (onursumer) [reply src/shared/components/lazyMobXTable/LazyMobXTable.tsx:253]: @alisman @adamabeshouse according to my comment [here](https://github.com/cBioPortal/cbioportal-frontend/pull/1232/files#diff-c2cf724d243fce1017b7b24de3932de8R197) we were not observing that variable anywhere. I guess it is okay to make it observable as long as we don't end up with infinite rendering or performance issues.

**Pull request merged**